esmvalcore/preprocessor/_derive/vegfrac.py

```diff
@@ -0,0 +1,22 @@
+"""Derivation of variable `vegFrac`."""
+
+import iris
+from ._baseclass import DerivedVariableBase
+
+
+class DerivedVariable(DerivedVariableBase):
+    """Derivation of variable `vegFrac`."""
+
+    # Required variables
+    required = [{'short_name': 'baresoilFrac', }]
+
+    @staticmethod
+    def calculate(cubes):
+        """Compute vegetation fraction from bare soil fraction.
+
+        """
+        baresoilfrac_cube = cubes.extract_strict(
+            iris.Constraint(name='area_fraction'))
+
+        baresoilfrac_cube.data = 1. - baresoilfrac_cube.data
```

Comment:
@mattiarighi this is the only like that manipulates the data in this derivation. I suspect that something this simple shouldn't realise the data, right?

Reply:
Not sure. @schlunma what do you think?

Reply:
Looking at point 6.3 from https://scitools.org.uk/iris/docs/latest/userguide/real_and_lazy_data.html

I would guess that this is not lazy. However, if we change it to this suggestion, it should be okay.
```suggestion
        baresoilfrac_cube= 1. - baresoilfrac_cube
```

Reply:
and that returns a dask array, so you can just do `baresoilfrac_cube.core_data() = 1. - baresoilfrac_cube.core_data()` and to realize it, `baresoilfrac_cube.compute()`

Reply:
So you're suggesting:
```suggestion
        baresoilfrac_cube.data() = 1. - baresoilfrac_cube.data()
        baresoilfrac_cube.compute()
```

Reply:
Looking at other derivation functions (e.g. [rlns](https://github.com/ESMValGroup/ESMValCore/blob/development/esmvalcore/preprocessor/_derive/rlns.py)), it looks like you simply need:
```
baresoilfrac_cube = 1. - baresoilfrac_cube
```

Reply:
Ah, but that's not supported, crap.

Reply:
In [_shared.py](https://github.com/ESMValGroup/ESMValCore/blob/development/esmvalcore/preprocessor/_derive/_shared.py) there are similar operations.

Reply:
sorry my line is `baresoilfrac_cube.data = 1. - baresoilfrac_cube.core_data()` - note that calling `.data` here keeps things lazy and the operation is done using the dask array `core_data()`

Reply:
I found that that doesn't work, but this weird hack does seem to work:
```
        baresoilfrac_cube = -1. * baresoilfrac_cube + 1.
```

Reply:
hahaha, it works of course, this is a hilarious implementation man :grin:

Reply:
@mattiarighi yeah, also not the way to do it :imp: